### ISSUE TITLE
[BUG] <title> no gradient when only tune encoder part.

### ISSUE THREAD

**Issue by YaxinLi0-0** [OpenBMB/MiniCPM-V#637]
### 是否已有关于该错误的issue或讨论？ | Is there an existing issue / discussion for this?

- [X] 我已经搜索过已有的issues和讨论 | I have searched the existing issues / discussions

### 该问题是否在FAQ中有解答？ | Is there an existing answer for this in FAQ?

- [X] 我已经搜索过FAQ | I have searched FAQ

### 当前行为 | Current Behavior

Hi,

I'm trying to freeze the llm part and tune the vision encoder and resampler with the finetune.py script. While I encounter the problem "Leaf variable was used in an inplace operation" and the error is located at 'modeling_minicpmv.py' line 166:

cur_vllm_emb += cur_vs_hs[0].mean() * 0
 
I change it to:

cur_vllm_emb = cur_vllm_emb + cur_vs_hs[0].mean() * 0

so the error disappear. I also manually set llm part requires_grid to false and the vpm part requires_grid to true. However, during the training, the grad_norm is consistently 0, nothing is updated.

I would be appreciate if you could provide any possible reason that cause this problem or directions to solve the problem.

Looking forward to your attention to this matter.

### 运行环境 | Environment

```Markdown
- OS:
- Python:
- Transformers:
- PyTorch:
- CUDA (`python -c 'import torch; print(torch.version.cuda)'`):
```
